### DISCUSSION TITLE
Automatically adjust AR/OD when using DT/HT mods with Difficulty Adjust

### DISCUSSION THREAD

**Discussion by vatei** [ppy/osu#30014] [Ideas]
Ignoring speed adjust changes in DA was previously suggested here, as a toggle https://github.com/ppy/osu/discussions/15249, but I believe that Difficulty Adjust mod should always set the AR/OD values defined by the user (or at the very least that should be the default behavior)

Furthermore, the discussion mentioned that implementing this would require a significant engine rewrite. 
That's why I'm instead suggesting an approach that would automate the compensation of AR/OD values in the background. This mimics what players are already doing manually (adjusting the slider until it matches the intended values) without requiring engine changes.

**Comment by Givikap120** [comment]
This is good idea but sadly it's VERY hard to do
There were multiple solutions:
- add "preserve on rate-change" switch on DA
- add "don't change AR/OD" switch on DT/HT
- "automatically adjust DA setting" global option

All are bad because of multiple reasons
First one (or the one you proposed) is bad because it's completely destroying the structure of mods. Mods can't be dependent on other mods. And such fundamental fact won't be changed
